theano/tensor/nlinalg.py

```diff
@@ -292,7 +289,7 @@ def grad(self, inputs, g_outputs):
         return [gz * self(x) * matrix_inverse(x).T]
 
     def infer_shape(self, node, shapes):
-        return [()]
+        return [shapes[0]]
```

Comment:
Hello, I recently bumped into this problem when trying to translate an algorithm implementation from Numpy to Theano, and I see there is a bit of a misunderstanding here.

Numpy's linear algebra package considers any N D array as an N-2 D array of 2D matrices. So for an array of shape N1×N2×N3×...×Nn×M×M the numpy determinant function would return the results for each M×M matrix in an array of size N1×N2×N3×...×Nn.

Link to the relevant numpy documentation: https://docs.scipy.org/doc/numpy-1.12.0/reference/routines.linalg.html#linear-algebra-on-several-matrices-at-once